### PR TITLE
redefine restart policy

### PR DESCRIPTION
**Pull request by CMGS**

### CODE REVIEW

File: engine/systemd/unit.go [hunk 1]
```diff
@@ -221,18 +216,11 @@ func (b *unitBuilder) buffer() (*bytes.Buffer, error) {
...
-	case strings.HasPrefix(restart, "on-failure"):
-		policy = "on-failure"
-	default:
-		err = fmt.Errorf("restart policy not supported: %s", restart)
+func (b *unitBuilder) convertToSystemdRestartPolicy(restart int) (policy string) {
+	if restart > 0 {
+		return "always"
 	}
-	return
+	return "no"
```
Comment (jasonjoo2010) [inline engine/systemd/unit.go]: Due to `No description provided.`, so is infinite restarting still supported? How to set it if so?

File: engine/docker/container.go [hunk 2]
```diff
@@ -165,7 +165,8 @@ func (e *Engine) VirtualizationCreate(ctx context.Context, opts *enginetypes.Vir
 		},
 		NetworkMode: networkMode,
 		RestartPolicy: dockercontainer.RestartPolicy{
-			Name: opts.RestartPolicy,
+			Name: restartPolicy,
+			// compatible with old days
 			// MaximumRetryCount: restartRetryCount,
```
Comment (jasonjoo2010) [inline engine/docker/container.go]: You just mentioned that the retry count is meaningful but where is it now?

File: engine/systemd/unit.go (hunk 1 above)
Comment (jasonjoo2010) [reply engine/systemd/unit.go]: Or limited retrying is not supported in `systemd`?

**Review by jasonjoo2010 — CHANGES_REQUESTED** [review]

File: engine/docker/container.go (hunk 2 above)
Comment (jschwinger233) [reply engine/docker/container.go]: 把 opts.Restart 传给 MaximumRetryCount 就好了

Comment (CMGS) [reply engine/docker/container.go]: @jschwinger23  -1 为永久 0 为关 >0 才有 retrycount 的效果，改了下你看看

File: engine/systemd/unit.go [hunk 3]
```diff
@@ -221,18 +216,11 @@ func (b *unitBuilder) buffer() (*bytes.Buffer, error) {
...
-	case strings.HasPrefix(restart, "on-failure"):
-		policy = "on-failure"
-	default:
-		err = fmt.Errorf("restart policy not supported: %s", restart)
+func (b *unitBuilder) convertToSystemdRestartPolicy(restart int) (policy string) {
+	if restart == -1 {
+		return "always"
 	}
-	return
+	return "no"
```
Comment (jasonjoo2010) [inline engine/systemd/unit.go]: So it will return `no` with restart = 10. Is it expected?

**Review by jasonjoo2010 — CHANGES_REQUESTED** [review]

**Review by jschwinger233 — APPROVED** [review]

**Pull request merged**